cli/pkg/kctrl/cmd/app/init/init.go

```diff
@@ -0,0 +1,336 @@
+}
+
+func printInformation(ui cmdcore.AuthoringUI) {
+	ui.PrintInformationalText("\n**Information**\napp-build.yml is generated as part of this flow. This file can be used for further updating and adding complex scenarios while using the `kctrl dev deploy` command. Please read the link'ed documentation for more explanation.")
+}
+
+func (createStep CreateStep) createAppFromAppBuild() kcv1alpha1.App {
+	//TODO Should we ask for the app Name ?
+	appName := "microservices-demo"
+	serviceAccountName := fmt.Sprintf("sa-%s", appName)
+	appAnnotation := map[string]string{
+		fetch.LocalFetchAnnotationKey: ".",
+	}
+	appTemplateSection := createStep.appBuild.Spec.App.Spec.Template
+	//TODO should we remove the fetch section as it is not beind used and add it dynamically during dev deploy.
+	appFetchSection := []kcv1alpha1.AppFetch{kcv1alpha1.AppFetch{
+		HTTP: &kcv1alpha1.AppFetchHTTP{},
+	}}
+	appDeploySection := createStep.appBuild.Spec.App.Spec.Deploy
+	return kcv1alpha1.App{
+		TypeMeta: v1.TypeMeta{
+			APIVersion: "kappctrl.k14s.io/v1alpha1",
+			Kind:       "App",
+		},
+		ObjectMeta: v1.ObjectMeta{
+			Name:        appName,
+			Annotations: appAnnotation,
+		},
+		Spec: kcv1alpha1.AppSpec{
+			ServiceAccountName: serviceAccountName,
+			Fetch:              appFetchSection,
+			Template:           appTemplateSection,
+			Deploy:             appDeploySection,
+		},
+	}
+}
+
+func (createStep CreateStep) updateExistingApp() (kcv1alpha1.App, error) {
+	var existingApp kcv1alpha1.App
+	data, err := os.ReadFile(AppFileName)
+	if err != nil {
+		return kcv1alpha1.App{}, err
+	}
+	err = yaml.Unmarshal(data, &existingApp)
+	if err != nil {
+		return kcv1alpha1.App{}, err
+	}
+	fetchSource := createStep.appBuild.ObjectMeta.Annotations[fetch.FetchContentAnnotationKey]
+	if fetchSource == fetch.FetchFromLocalDirectory {
+		return existingApp, nil
+	}
+
+	templateSectionFromExistingApp := existingApp.Spec.Template
+
+	//As fetchSource is not from Local directory, we should add upstream folder in ytt path and helmTemplate(if required).
+	isHelmTemplateRequired := false
+	if fetchSource == fetch.FetchChartFromHelmRepo || fetchSource == fetch.FetchChartFromGithub {
+		isHelmTemplateRequired = true
+	}
+	if isHelmTemplateRequired {
+		//TODO Figure out the edge scenarios
+		addUpstreamAsPathToHelmIfNotExist(templateSectionFromExistingApp)
+		addStdInAsPathToYttIfNotExist(templateSectionFromExistingApp)
+
+	}
+
+	addUpstreamAsPathToYttIfNotExist(templateSectionFromExistingApp)
+
+	return existingApp, nil
+}
+
+func (createStep CreateStep) configureExportSection() {
+	fetchSource := createStep.appBuild.ObjectMeta.Annotations[fetch.FetchContentAnnotationKey]
+	if fetchSource == fetch.FetchFromLocalDirectory {
+		return
+	}
+
+	// TODO current implementation is if export section is already defined, we will not touch it. Confirm the same.
+	if createStep.appBuild.Spec.Export == nil || len(createStep.appBuild.Spec.Export) == 0 {
```

Comment:
If user has sync'ed vendir configuration from somewhere and then in the next run, they remove the upstream folder from includePaths, should we add it or leave it removed? Similarly, If they move to LocalDirectory in the next run, we should remove it, right?

Reply:
We are not gonna allow transition from one fetch source to another.